src/azure-cli-core/azure/cli/core/auth/identity.py

```diff
@@ -196,7 +192,7 @@ def login_with_service_principal(self, client_id, credential, scopes):
         """
         sp_auth = ServicePrincipalAuth.build_from_credential(self.tenant_id, client_id, credential)
         client_credential = sp_auth.get_msal_client_credential()
-        cca = ConfidentialClientApplication(client_id, client_credential, **self._msal_app_kwargs)
+        cca = ConfidentialClientApplication(client_id, client_credential=client_credential, **self._msal_app_kwargs)
```

Comment:
`ConfidentialClientApplication`'s `client_credential` is a keyword argument. #29439 passes a positional argument, breaking tests such as `azure.cli.core.auth.tests.test_identity.TestIdentity.test_login_with_service_principal_certificate`.

Also see https://github.com/Azure/azure-cli-dev-tools/issues/318